### PR TITLE
amrex: fix handling of CUDA arch

### PR DESCRIPTION
**Pull request by mic84**

### CODE REVIEW

**Comment by eschnett** [comment]
This pull request does not work. Try this patch:
```diff
$ git diff
diff --git a/var/spack/repos/builtin/packages/amrex/package.py b/var/spack/repos/builtin/packages/amrex/package.py
index 37c8a9838..9d2403c3f 100644
--- a/var/spack/repos/builtin/packages/amrex/package.py
+++ b/var/spack/repos/builtin/packages/amrex/package.py
@@ -121,7 +121,7 @@ def url_for_version(self, version):
             url = "https://github.com/AMReX-Codes/amrex/archive/{0}.tar.gz"
         return url.format(version.dotted)

-    def get_cuda_arch_string(values):
+    def get_cuda_arch_string(self, values):
         if 'none' in values:
             return('Auto')
         else:
@@ -160,8 +160,8 @@ def cmake_args(self):
             args.append('-DCMAKE_Fortran_MODDIR_FLAG=-M')

         if '+cuda' in self.spec:
-            cuda_arch = spec.variants['cuda_arch'].value
-            args.append('-DCUDA_ARCH=' + get_cuda_arch_string(cuda_arch))
+            cuda_arch = self.spec.variants['cuda_arch'].value
+            args.append('-DCUDA_ARCH=' + self.get_cuda_arch_string(cuda_arch))

         return args

@@ -197,7 +197,7 @@ def cmake_args(self):
             args.append('-DAMReX_GPU_BACKEND=CUDA')
             args.append('-DAMReX_CUDA_ERROR_CAPTURE_THIS=ON')
             args.append('-DAMReX_CUDA_ERROR_CROSS_EXECUTION_SPACE_CALL=ON')
-            cuda_arch = spec.variants['cuda_arch'].value
-            args.append('-DCUDA_ARCH=' + get_cuda_arch_string(cuda_arch))
+            cuda_arch = self.spec.variants['cuda_arch'].value
+            args.append('-DCUDA_ARCH=' + self.get_cuda_arch_string(cuda_arch))

         return args
```

**Comment by mic84** [comment]
@eschnett good catch! Thanks! Now it should be good to go. Can you confirm?

**Comment by eschnett** [comment]
I see no difference between my working working tree and your pull request. It seems good to go.

File: var/spack/repos/builtin/packages/amrex/package.py [hunk 1]
```diff
@@ -121,6 +121,14 @@ def url_for_version(self, version):
             url = "https://github.com/AMReX-Codes/amrex/archive/{0}.tar.gz"
         return url.format(version.dotted)
 
+    def get_cuda_arch_string(self, values):
+        if 'none' in values:
+            return('Auto')
+        else:
+            # Use format x.y instead of CudaPackage xy format
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/amrex/package.py:128]: Oh wow, I looked at this and was sure xy is also what we pass. Well, we don't :)

**Review by ax3l — APPROVED** [review]

File: var/spack/repos/builtin/packages/amrex/package.py [hunk 2]
```diff
@@ -121,6 +121,14 @@ def url_for_version(self, version):
             url = "https://github.com/AMReX-Codes/amrex/archive/{0}.tar.gz"
         return url.format(version.dotted)
 
+    def get_cuda_arch_string(self, values):
+        if 'none' in values:
+            return('Auto')
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/amrex/package.py]:
```suggestion
            return 'Auto'
```

File: var/spack/repos/builtin/packages/amrex/package.py [hunk 3]
```diff
@@ -121,6 +121,14 @@ def url_for_version(self, version):
...
+        else:
+            # Use format x.y instead of CudaPackage xy format
+            vf = tuple(float(x) / 10.0 for x in values)
+            return(';'.join(str(x) for x in vf))
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/amrex/package.py]:
```suggestion
            return ';'.join(str(x) for x in vf)
```

**Review by ax3l — COMMENTED** [review]
Added two comments that look like curious python

**Review by ax3l — APPROVED** [review]

**Pull request merged**